Mathlib/Algebra/Homology/DerivedCategory/Ext.lean

```diff
@@ -303,8 +317,107 @@ lemma homAddEquiv_apply (α : Ext X Y n) : homAddEquiv α = α.hom := rfl
 
 end
 
+/-- The postcomposition `Ext X Y a →+ Ext X Z b` with `β : Ext Y Z n` when `a + n = b`. -/
+@[simps!]
+noncomputable def postcomp (β : Ext Y Z n) (X : C) {a b : ℕ} (h : a + n = b) :
+    Ext X Y a →+ Ext X Z b :=
+  AddMonoidHom.mk' (fun α => α.comp β h) (by simp)
+
+/-- The precomposition `Ext Y Z a →+ Ext X Z b` with `α : Ext X Y n` when `n + a = b`. -/
+@[simps!]
+noncomputable def precomp (α : Ext X Y n) (Z : C) {a b : ℕ} (h : n + a = b) :
+    Ext Y Z a →+ Ext X Z b :=
+  AddMonoidHom.mk' (fun β => α.comp β h) (by simp)
+
 end Ext
 
 end Abelian
 
+open Abelian
+
+namespace ShortComplex
+
+variable (S : ShortComplex C)
+
+lemma ext_mk₀_f_comp_ext_mk₀_g : (Ext.mk₀ S.f).comp (Ext.mk₀ S.g) (zero_add 0) = 0 := by simp
+
+namespace ShortExact
+
+variable {S}
+variable (hS : S.ShortExact)
+
+/-- The class in `Ext S.X₃ S.X₁ 1` that is attached to a short exact
+short complex `S` in an abelian category. -/
+noncomputable def extClass : Ext.{w} S.X₃ S.X₁ 1 := by
+  let W := (HomologicalComplex.quasiIso C (ComplexShape.up ℤ))
+  let S' := S.map (CochainComplex.singleFunctor C 0)
+  have hS' : S'.ShortExact := hS.map_of_exact (HomologicalComplex.single _ _ _)
+  let K := CochainComplex.mappingCone S'.f
+  let w : K ⟶ S'.X₃ := CochainComplex.mappingCone.descShortComplex S'
+  have hw : W w := CochainComplex.mappingCone.quasiIso_descShortComplex hS'
+  let δ : K ⟶ S'.X₁⟦(1 : ℤ)⟧ := (CochainComplex.mappingCone.triangle S'.f).mor₃
+  have : HasSmallLocalizedShiftedHom.{w} W ℤ S'.X₃ S'.X₁ := by
+    dsimp [W, S']
+    infer_instance
+  have : HasSmallLocalizedHom.{w} W S'.X₃ K := by
+    rw [Localization.hasSmallLocalizedHom_iff_target W S'.X₃ w hw]
+    dsimp [W, S']
+    apply Localization.hasSmallLocalizedHom_of_hasSmallLocalizedShiftedHom₀ (M := ℤ)
+  have : HasSmallLocalizedShiftedHom.{w} W ℤ K S'.X₁ := by
+    rw [Localization.hasSmallLocalizedShiftedHom_iff_source.{w} W ℤ w hw S'.X₁]
+    dsimp [W, S']
+    infer_instance
```

Comment:
Is it possible to promote these to instances?

Reply:
No, we cannot, because the proofs depend on a property of `S'` (i.e. that it is short-exact).